contracts/supplementary/rebalancing-manager/RebalancingTokenManager.sol

```diff
@@ -0,0 +1,456 @@
+            "RebalancingTokenManager.proposeNewRebalance: State must be in Default"
+        );
+
+        // Get price data
+        (
+            uint256 btcPrice,
+            uint256 ethPrice
+        ) = queryPriceData();
+
+        // Require that allocation has changed sufficiently enough to justify rebalance
+        uint256 currentSetDollarAmount = checkSufficientAllocationChange(
+            btcPrice,
+            ethPrice,
+            rebalancingSetInterface.currentSet()
+        );
+        
+        // Create new Set Token that collateralizes Rebalancing Set Token
+        (
+            address nextSetAddress,
+            uint256 auctionStartPrice,
+            uint256 auctionPivotPrice
+        ) = createNewAllocationSetToken(
+            btcPrice,
+            ethPrice,
+            currentSetDollarAmount
+        );
+
+        // Propose new allocation to Rebalancing Set Token
+        rebalancingSetInterface.propose(
+            nextSetAddress,
+            auctionLibrary,
+            auctionTimeToPivot,
+            auctionStartPrice,
+            auctionPivotPrice
+        );
+    }
+
+    /* ============ Internal ============ */
+    /*
+     * Query price feeds for wbtc and weth, return as uint256
+     *
+     */
+    function queryPriceData()
+        private
+        view
+        returns (uint256, uint256)
+    {
+        // Get prices from oracles
+        bytes32 btcPriceBytes = btcPriceFeed.read();
+        bytes32 ethPriceBytes = ethPriceFeed.read();
+
+        // Cast bytes32 prices to uint256
+        uint256 btcPrice = uint256(btcPriceBytes);
+        uint256 ethPrice = uint256(ethPriceBytes);
+
+        return(btcPrice, ethPrice);
+    }
+
+    /*
+     * Check there has been a sufficient change in allocation (greater than 2%) and return
+     * USD value of currentSet.
+     *
+     * @param  _btcPrice              The 18 decimal value of one full BTC
+     * @param  _ethPrice              The 18 decimal value of one full ETH
+     * @param  _currentSetAddress     The address of the Rebalancing Set Token's currentSet
+     * @return                        The currentSet's USD value (in cents)
+     */
+    function checkSufficientAllocationChange(
+        uint256 _btcPrice,
+        uint256 _ethPrice,
+        address _currentSetAddress
+    )
+        private
+        view
+        returns (uint256)
+    {
+        // Create current set interface
+        ISetToken currentSetTokenInterface = ISetToken(_currentSetAddress);
+
+        // Get naturalUnit and units of currentSet
+        uint256 currentSetNaturalUnit = currentSetTokenInterface.naturalUnit();
+        uint256[] memory currentSetUnits = currentSetTokenInterface.getUnits();
+
+        // Calculate wbtc dollar value in currentSet (in cents)
+        uint256 btcDollarAmount = calculateTokenAllocationAmountUSD(
+            _btcPrice,
+            currentSetNaturalUnit,
+            currentSetUnits[0],
```

Comment:
This means btc always has to be the first component/unit stored?

Reply:
Yeah, we have to get that right on deployment, but after that it's self-reinforcing